### PR TITLE
allow html without \n after

### PR DESCRIPTION
**Pull request by UziTech**
**Marked version:** master

**Markdown flavor:** Markdown.pl|CommonMark|GitHub Flavored Markdown|n/a

## Description

Fix html without a new line after

```
marked('<img src="sdfg">\n');
// expected: '<img src="sdfg">'
// actual:   '<img src="sdfg">'

marked('<img src="sdfg">');
// expected: '<img src="sdfg">'
// actual:   '<p><img src="sdfg"></p>'
```

- Fixes https://github.com/markedjs/marked/issues/1232#issuecomment-470663325

## Contributor

- [x] Test(s) exist to ensure functionality and minimize regression

## Committer

In most cases, this should be a different person than the contributor.

- [x] Draft GitHub release notes have been updated.
- [x] CI is green (no forced merge required).
- [x] Merge PR

### CODE REVIEW

**Comment by styfle** [comment]
There are some merge conflicts

File: lib/marked.js [hunk 1]
```diff
@@ -27,8 +27,8 @@ var block = {
     + '|<![A-Z][\\s\\S]*?>\\n*' // (4)
     + '|<!\\[CDATA\\[[\\s\\S]*?\\]\\]>\\n*' // (5)
     + '|</?(tag)(?: +|\\n|/?>)[\\s\\S]*?(?:\\n{2,}|$)' // (6)
-    + '|<(?!script|pre|style)([a-z][\\w-]*)(?:attribute)*? */?>(?=\\h*\\n)[\\s\\S]*?(?:\\n{2,}|$)' // (7) open tag
-    + '|</(?!script|pre|style)[a-z][\\w-]*\\s*>(?=\\h*\\n)[\\s\\S]*?(?:\\n{2,}|$)' // (7) closing tag
```
Comment (UziTech) [inline lib/marked.js]: @Feder1co5oave I'm not sure what the `\\h*` is supposed to do. As far as I know `\H` isn't a special character.

Comment (davisjam) [reply lib/marked.js]: `\h` has meaning in some languages.

| Language | Meaning of `\h` |
| ------------- | --------------- |
| Ruby | Hex chars, e.g. `[0-9a-fA-F]` |
| Java, PHP, Perl | Horizontal whitespace, cf. `\v` |

Comment (davisjam) [reply lib/marked.js]: ...so I imagine this regex originated in a language (or in someone's head) where `\h` had meaning.

Comment (davisjam) [reply lib/marked.js]: ...I've written a research paper on this topic if @UziTech @styfle @Feder1co5oave would be interested.

File: lib/marked.js [hunk 2]
```diff
@@ -27,8 +27,8 @@ var block = {
...
+    + '|<(?!script|pre|style)([a-z][\\w-]*)(?:attribute)*? */?>(?=\\n|$)[\\s\\S]*?(?:\\n{2,}|$)' // (7) open tag
```
Comment (davisjam) [inline lib/marked.js]: - Is there a test case that includes horizontal whitespace where the `\h` was?
- It's possible the original regex should have used something like `[ \t]` instead?

File: lib/marked.js (hunk 1 above)
Comment (davisjam) [reply lib/marked.js]: - Is there a test case that includes horizontal whitespace where the \h was?
- It's possible the original regex should have used something like [ \t] instead?

Comment (UziTech) [reply lib/marked.js]: It looks like it should be `[ \\t]*` but I would like to get @Feder1co5oave point of viewq since he is the one that introduced it.

Comment (Feder1co5oave) [reply lib/marked.js]: It was intended to be `[ \t]*` I guess, I just don't remember why I used `\h` instead, and why it required a newline after the tag :(

Comment (UziTech) [reply lib/marked.js]: I added `[ \t]*` in place of `\h` but with some manual testing it doesn't seem to change anything. All of the tests I can think of still pass with or without `[ \t]*`

Comment (UziTech) [reply lib/marked.js]: That could be due to the fact that we strip white space before we check the output.

**Comment by UziTech** [comment]
@davisjam @joshbruce This should be ready to merge. Once this is merged we can release [v0.6.2](https://github.com/markedjs/marked/pull/1441)

File: lib/marked.js (hunk 1 above)
Comment (Feder1co5oave) [reply lib/marked.js]: We still strip whitespace when testing? Wasn't html-differ supposed to fix this?

Comment (UziTech) [reply lib/marked.js]: We were only using html-differ for the spec tests. The files were still using the manual method.

I fixed it in #1449

**Comment by UziTech** [comment]
@davisjam @joshbruce We need one of you to approve this in order to merge it and release a new version of marked.

**Comment by joshbruce** [comment]
Sorry. I'm gonna be a little slow on things for at least the next few days (family-related keeping me offline for long stretches). Maybe we should consider an alternative to keep from deadlock. Though not sure it will happen often.

**Pull request merged**